server/src/main/java/io/druid/server/http/CoordinatorCompactionConfigsResource.java

```diff
@@ -0,0 +1,209 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.http;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.sun.jersey.spi.container.ResourceFilters;
+import io.druid.audit.AuditInfo;
+import io.druid.audit.AuditManager;
+import io.druid.common.config.JacksonConfigManager;
+import io.druid.server.coordinator.CoordinatorCompactionConfig;
+import io.druid.server.coordinator.DataSourceCompactionConfig;
+import io.druid.server.http.security.ConfigResourceFilter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Path("/druid/coordinator/v1/config/compaction")
+@ResourceFilters(ConfigResourceFilter.class)
+public class CoordinatorCompactionConfigsResource
+{
+  private final JacksonConfigManager manager;
+
+  @Inject
+  public CoordinatorCompactionConfigsResource(JacksonConfigManager manager)
+  {
+    this.manager = manager;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getCompactConfig()
+  {
+    return Response.ok(
+        manager.watch(
+            CoordinatorCompactionConfig.CONFIG_KEY,
+            CoordinatorCompactionConfig.class
+        ).get()
+    ).build();
+  }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response setCompactionTaskLimit(
+      @QueryParam("slotRatio") Double compactionTaskSlotRatio,
+      @QueryParam("maxSlots") Integer maxCompactionTaskSlots,
+      @HeaderParam(AuditManager.X_DRUID_AUTHOR) @DefaultValue("") final String author,
+      @HeaderParam(AuditManager.X_DRUID_COMMENT) @DefaultValue("") final String comment,
+      @Context HttpServletRequest req
+  )
+  {
+    CoordinatorCompactionConfig current = manager.watch(
+        CoordinatorCompactionConfig.CONFIG_KEY,
+        CoordinatorCompactionConfig.class
+    ).get();
+
+    final CoordinatorCompactionConfig newCompactionConfig;
+    if (current != null) {
+      newCompactionConfig = CoordinatorCompactionConfig.from(current, compactionTaskSlotRatio, maxCompactionTaskSlots);
+    } else {
+      newCompactionConfig = new CoordinatorCompactionConfig(null, compactionTaskSlotRatio, maxCompactionTaskSlots);
+    }
+
+    if (!manager.set(
+        CoordinatorCompactionConfig.CONFIG_KEY,
+        newCompactionConfig,
+        new AuditInfo(author, comment, req.getRemoteAddr())
+    )) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
```

Comment:
minor nit: is it possible to add some message about the failure cause here ? Maybe something on lines - Failed to set config...

Reply:
Added.